### PR TITLE
Add basic test for podman-quadlet

### LINKED ISSUE #998: Include new `podman` with `quadlet` functionality
Please try to answer the following questions about the package you are requesting:

1. **What, if any, are the additional dependencies on the package? (i.e. does it pull in Python, Perl, etc)**
Requires `podman` and `crun`, both of which are provided in base already.

2. **What is the size of the package and its dependencies?**
Unsure, though likely less that 50kB.

3. **What problem are you trying to solve with this package? Or what functionality does the package provide?**
Integrating Podman with systemd requires that either `podman generate systemd` is used (which presupposes a Podman container, isn't as configurable, and produces files that aren't intended for further editing), or that equivalent integration is done manually and in a repeated manner. Quadlet solves the issue of integration with systemd as a starting point in a much more elegant way.

4. **Can the software provided by the package be run from a container?  Explain why or why not.**
This is intended to be used on the base system itself, and generates service files that themselves orchestrate Podman containers.

5. **Can the tool(s) provided by the package be helpful in debugging container runtime issues?**
Likely not, though they will likely help alleviate issues with misconfigured integrations.

6. **Can the tool(s) provided by the package be helpful in debugging networking issues?**
I don't believe so.

7. **Is it possible to layer the package onto the base OS as a day 2 operation?  Explain why or why not.**
It's possible, though unwieldy for such a small utility.

8. **In the case of packages providing services and binaries, can the packaging be adjusted to just deliver binaries?**
This will likely only provide a binary and related support files for systemd generators.

9. **Can the tool(s) provided by the package be used to do things we’d rather users not be able to do in FCOS? (e.g. can it be abused as a Turing complete interpreter?)**
I don't believe so.

10. **Does the software provided by the package have a history of CVEs?**
It does not, though it's a fairly young project.

### PR DESCRIPTION
**Pull request by travier**
Fixes: coreos/fedora-coreos-tracker#998

### CODE REVIEW

**Comment by travier** [comment]
Trying this, I get:

```
warning: Couldn't find locked package 'podman-5:4.4.0-1.fc37'
  Packages matching NEVRA: 0; Mismatched checksums: 0
  Packages matching name and arch (18):
  podman-3:4.0.1-2.fc36.x86_64 (fedora-coreos-pool)
  podman-3:4.0.1-2.fc37.x86_64 (fedora-coreos-pool)
  podman-3:4.0.2-1.fc36.x86_64 (fedora-coreos-pool)
  podman-3:4.0.3-1.fc36.x86_64 (fedora-coreos-pool)
  podman-3:4.1.0-1.fc36.x86_64 (fedora-coreos-pool)
  podman-3:4.1.0-8.fc36.x86_64 (fedora-coreos-pool)
  podman-3:4.1.1-1.fc36.x86_64 (fedora-coreos-pool)
  podman-3:4.1.1-2.fc36.x86_64 (fedora-coreos-pool)
  podman-4:4.1.1-3.fc36.x86_64 (fedora-coreos-pool)
  podman-4:4.2.0-2.fc36.x86_64 (fedora-coreos-pool)
  podman-4:4.2.0-2.fc37.x86_64 (fedora-coreos-pool)
  podman-4:4.2.1-2.fc36.x86_64 (fedora-coreos-pool)
  podman-4:4.2.1-2.fc37.x86_64 (fedora-coreos-pool)
  podman-4:4.3.0-2.fc36.x86_64 (fedora-coreos-pool)
  podman-4:4.3.0-2.fc37.x86_64 (fedora-coreos-pool)
  podman-4:4.3.1-1.fc37.x86_64 (fedora-coreos-pool)
  podman-4:4.2.1-2.fc37.x86_64 (fedora)
  podman-4:4.3.1-1.fc37.x86_64 (fedora-updates)
warning: Couldn't find locked package 'podman-quadlet-5:4.4.0-1.fc37'
  Packages matching NEVRA: 0; Mismatched checksums: 0
  No packages matched name: podman-quadlet arch: 
error: Packages not found: podman, podman-quadlet
```

Looks like it's not looking at the testing repo.

**Comment by dustymabe** [comment]
CI is failing with:

```
No such rpm: containers-common-1-78.fc37.x86_64
```

Maybe you need to fast-track other packages?

**Comment by dustymabe** [comment]
> Trying this, I get:
> 
> ```
> warning: Couldn't find locked package 'podman-5:4.4.0-1.fc37'
>   Packages matching NEVRA: 0; Mismatched checksums: 0
>   Packages matching name and arch (18):
>   podman-3:4.0.1-2.fc36.x86_64 (fedora-coreos-pool)
>   podman-3:4.0.1-2.fc37.x86_64 (fedora-coreos-pool)
>   podman-3:4.0.2-1.fc36.x86_64 (fedora-coreos-pool)
>   podman-3:4.0.3-1.fc36.x86_64 (fedora-coreos-pool)
>   podman-3:4.1.0-1.fc36.x86_64 (fedora-coreos-pool)
>   podman-3:4.1.0-8.fc36.x86_64 (fedora-coreos-pool)
>   podman-3:4.1.1-1.fc36.x86_64 (fedora-coreos-pool)
>   podman-3:4.1.1-2.fc36.x86_64 (fedora-coreos-pool)
>   podman-4:4.1.1-3.fc36.x86_64 (fedora-coreos-pool)
>   podman-4:4.2.0-2.fc36.x86_64 (fedora-coreos-pool)
>   podman-4:4.2.0-2.fc37.x86_64 (fedora-coreos-pool)
>   podman-4:4.2.1-2.fc36.x86_64 (fedora-coreos-pool)
>   podman-4:4.2.1-2.fc37.x86_64 (fedora-coreos-pool)
>   podman-4:4.3.0-2.fc36.x86_64 (fedora-coreos-pool)
>   podman-4:4.3.0-2.fc37.x86_64 (fedora-coreos-pool)
>   podman-4:4.3.1-1.fc37.x86_64 (fedora-coreos-pool)
>   podman-4:4.2.1-2.fc37.x86_64 (fedora)
>   podman-4:4.3.1-1.fc37.x86_64 (fedora-updates)
> warning: Couldn't find locked package 'podman-quadlet-5:4.4.0-1.fc37'
>   Packages matching NEVRA: 0; Mismatched checksums: 0
>   No packages matched name: podman-quadlet arch: 
> error: Packages not found: podman, podman-quadlet
> ```
> 
> Looks like it's not looking at the testing repo.

you need to stick the RPMs in `overrides/rpm` if you are testing this locally. Then `cosa fetch --with-cosa-overrides && cosa build`.

**Comment by travier** [comment]
Thanks, was missing the `--with-cosa-overrides` flag.

> CI is failing with:
> 
> ```
> No such rpm: containers-common-1-78.fc37.x86_64
> ```
> 
> Maybe you need to fast-track other packages?

According to https://bodhi.fedoraproject.org/updates/FEDORA-2023-20624cd618, it should even be in stable now.

File: tests/kola/containers/quadlet/config.bu [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    - path: /etc/containers/systemd/test.container
+      contents:
+        inline: |
+          [Unit]
+          Description=A minimal container
+          Before=local-fs.target
+
+          [Container]
+          # Use the centos image
+          Image=quay.io/centos/centos:latest
```
Comment (dustymabe) [inline tests/kola/containers/quadlet/config.bu]: Why CentOS here? Maybe something much smaller (`fedora-minimal` or even a tiny tiny `sleep` or `busybox` container?).

File: tests/kola/containers/quadlet/test.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#
+# Ensure that basic quadlet functionality works
+
+set -xeuo pipefail
+
+. $KOLA_EXT_DATA/commonlib.sh
+
+podman ps -a
+podman volume ls
+podman network ls
```
Comment (dustymabe) [inline tests/kola/containers/quadlet/test.sh]: Hmm. Shouldn't we evaluate the output of these commands to verify the state reflects what we expect it to be?

File: tests/kola/containers/quadlet/test.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+## kola:
+##   # We're pulling a container image from Quay.io
+##   tags: "platform-independent needs-internet"
```
Comment (dustymabe) [inline tests/kola/containers/quadlet/test.sh:4]: I think we do have other tests that pull containers that are `exclusive: false`. This one is probably on the fence of "making meaningful changes to the system" or not so I could go either way.

**Review by dustymabe — COMMENTED** [review]
Thank you for working on this Timothee!

**Comment by dustymabe** [comment]
> > CI is failing with:
> > ```
> > No such rpm: containers-common-1-78.fc37.x86_64
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Maybe you need to fast-track other packages?
> 
> According to https://bodhi.fedoraproject.org/updates/FEDORA-2023-20624cd618, it should even be in stable now.

Then remove it from the `manifest-lock.overrides.yaml` file.

**Comment by travier** [comment]
I'm starting with the examples from https://github.com/containers/podman/blob/main/docs/source/markdown/podman-systemd.unit.5.md#examples, which apparently don't fully work.

**Comment by jlebon** [comment]
To be safe, should we hold this until after promotion next week?

Edit: OK cool, had missed the convo in https://github.com/coreos/fedora-coreos-tracker/issues/998.

File: manifests/user-experience.yaml [hunk 4]
```diff
@@ -40,7 +40,7 @@ packages:
   ## crun recommends but doesn't require criu and criu-libs. We want them for
   ## checkpoint/restore. https://github.com/coreos/fedora-coreos-tracker/issues/1370
   - crun criu criu-libs
-  - podman
+  - podman podman-quadlet
```
Comment (dustymabe) [inline manifests/user-experience.yaml]: not sure if it is, but if this manifest is shared with RHCOS then we'll need to move `podman-quadlet` to a non-shared manifest for now.

File: tests/kola/containers/quadlet/test.sh (hunk 3 above)
Comment (travier) [reply tests/kola/containers/quadlet/test.sh:4]: Good question. We'll setup networks for containers and volumes. This should not interact with other things but the more container related tests we merge the more likely we'll have issues.

**Comment by travier** [comment]
https://github.com/containers/podman/pull/17354

**Comment by travier** [comment]
OK, next step would be to test pods.

File: manifests/fedora-coreos-base.yaml [hunk 5]
```diff
@@ -166,6 +166,8 @@ packages:
   - iptables-legacy
   # GPU Firmware files (not broken out into subpackage of linux-firmware in RHEL yet)
   - amd-gpu-firmware intel-gpu-firmware nvidia-gpu-firmware
+  # quadlet is currently not in RHEL and probably won't ever be in RHCOS
+  - podman-quadlet
```
Comment (LorbusChris) [inline manifests/fedora-coreos-base.yaml]: this won't be needed, as the podman-quadlet package split has been undone in https://github.com/containers/podman/pull/17351/files

**Review by LorbusChris — CHANGES_REQUESTED** [review]

File: manifests/fedora-coreos-base.yaml (hunk 5 above)
Comment (dustymabe) [reply manifests/fedora-coreos-base.yaml]: Interesting.. Can we get that backported to a F37 RPM in F37 so we don't have to wait for another release?

**Comment by travier** [comment]
```
[2023-02-10T16:37:58.026Z] --- PASS: ext.config.containers.quadlet (26.24s)
```

**Comment by dustymabe** [comment]
The upgrade test in CI failed because of a signature issue that I'm aware of and will fix. For now we can ignore the failure.

**Review by dustymabe — APPROVED** [review]
LGTM

**Pull request merged**